libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts

```diff
@@ -1,23 +1,55 @@
 import { PlaybackStore } from '@angular-spotify/web/shared/data-access/store';
-import { ChangeDetectionStrategy, Component } from '@angular/core';
-import { map } from 'rxjs/operators';
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  Input,
+  NgZone,
+  ViewChild
+} from '@angular/core';
+import { filter, map } from 'rxjs/operators';
 
 @Component({
   selector: 'as-album-art-overlay',
   templateUrl: './album-art-overlay.component.html',
   styleUrls: ['./album-art-overlay.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class AlbumArtOverlayComponent {
-  albumBackgroundImage$ = this.playbackStore.currentTrack$.pipe(
-    map((track) => {
-      if (!track?.album?.images) {
-        return null;
+export class AlbumArtOverlayComponent implements AfterViewInit {
+  @Input() set imageUrl(url: string) {
+    this.zone.runOutsideAngular(() => {
+      this.drawImageToCanvas(url);
+    });
+  }
+
+  @ViewChild('canvas') canvas!: ElementRef<HTMLCanvasElement>;
```

Comment:
suggestion: `canvas` can be queried with `static: true` so it is available in `ngOnInit` instead.